### PR TITLE
Added awesome lint and awesome bot Github Workflow Actions

### PR DESCRIPTION
**Pull request by anthonyronda**
[Awesome Lint](https://github.com/sindresorhus/awesome-lint) enforces the [Awesome List style guide](https://github.com/sindresorhus/awesome/blob/main/pull_request_template.md). Check the awesome lint readme for instructions on how to ignore rules for certain lines

[Awesome Bot](https://github.com/dkhamsing/awesome_bot) checks for dead links which is generally useful for Awesome Lists

### CODE REVIEW

**Comment by anthonyronda** [comment]
Turns out I was using a deprecated version of the GitHub Actions ruby environment for the awesome_bot gem. It now should run without the deprecation message.

I forgot to mention that I first found these on the [Awesome Firebase List](https://github.com/jthegedus/awesome-firebase)

If you have any other questions about these two ask away!

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+name: lint
+on:
+    pull_request:
+    push:
+      branches:
+        - main
```
Comment (bagelbits) [inline .github/workflows/lint.yaml]:
```suggestion
    pull_request:
      branches:
        - main
    push:
      branches:
        - main
```

Comment (bagelbits) [reply .github/workflows/lint.yaml]: This will also have it run on PRs.

**Review by bagelbits — APPROVED** [review]
So. I'm not sure it's actually working but I'll merge it in anyways to see.

**Pull request merged**